extensions/src/main/java/org/pentaho/platform/plugin/services/email/EmailService.java

```diff
@@ -186,18 +207,51 @@ protected PasswordAuthentication getPasswordAuthentication() {
       msg.setText( messages.getString( "EmailService.MESSAGE" ) );
       msg.setHeader( "X-Mailer", "smtpsend" );
       msg.setSentDate( new Date() );
-      Transport.send( msg );
-      sendEmailMessage = "EmailTester.SUCESS";
+      sendEmail( emailConfig, session, msg );
+      sendEmailMessage = TEST_EMAIL_SUCCESS;
     } catch ( Exception e ) {
       logger.error( messages.getString( "EmailService.NOT_CONFIGURED" ), e );
-      sendEmailMessage = "EmailTester.FAIL";
+      sendEmailMessage = TEST_EMAIL_FAIL;
+    }
+    if ( EmailConstants.GRANT_TYPE_AUTH_CODE.equals( emailConfig.getGrantType() ) ) {
+      setEmailConfig( emailConfig );
     }
     return sendEmailMessage;
   }
 
+  public void sendEmail( final IEmailConfiguration emailConfig, Session session, MimeMessage msg ) throws EmailServiceException {
+    try {
+      if ( emailConfig.getAuthMechanism().equals( EmailConstants.AUTH_TYPE_XOAUTH2 ) ) {
+        IEmailAuthenticationResponse token = getOAuthToken( emailConfig );
+        if ( emailConfig.getGrantType().equals( EmailConstants.GRANT_TYPE_AUTH_CODE ) ) {
+          emailConfig.setGrantType( EmailConstants.GRANT_TYPE_REFRESH_TOKEN );
+          emailConfig.setRefreshToken( token.getRefreshToken() );
+          emailConfig.setAuthorizationCode( "" );
+        }
+        if ( emailConfig.getSmtpProtocol().equals( EmailConstants.PROTOCOL_GRAPH_API ) ) {
+          ByteArrayOutputStream os = new ByteArrayOutputStream();
+          msg.writeTo( os );
+          String s = Base64.getEncoder().encodeToString( os.toByteArray() );
+          sendMailGraphApi( emailConfig, token.getAccessToken(), s  );
+        } else {
+          SMTPTransport transport = new SMTPTransport( session, null );
```

Comment:
No further action.

I think i found out how to configure SMTP Oauth without creating the SMTPTransport object directly : https://stackoverflow.com/questions/72498780/javax-mail-office-exchange-with-oauth2-token-smtp

```
props.put("mail.smtp.port", "587");
    props.put("mail.smtp.host", "smtp.office365.com");
    props.put("mail.smtp.auth.mechanisms", "XOAUTH2");
    props.put("mail.smtp.starttls.enable", "true");
    props.put("mail.smtp.starttls.required", "true");
    props.put("mail.smtp.auth", "true");
```
then
```
Session session = Session.getInstance(props);
    try {
        Transport transport = session.getTransport("smtp");
        transport.connect("smtp.office365.com", emailAddress, accessToken);
    } catch (MessagingException ex) {
        ex.printStackTrace();
    } 
   ```